xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2647,7 +2651,51 @@ void CVideoPlayer::OnExit()
     CLog::Log(LOGINFO, "VideoPlayer: eof, waiting for queues to empty");
 
   CFileItem fileItem(m_item);
-  UpdateFileItemStreamDetails(fileItem);
+  if (m_isBluray)
+  {
+    // Save playlist currently playing
+    SavePlaylistInformation();
+
+    // With BD-J discs everything played through the menu is indistinguishable from the menu itself
+    // So assume only the first playlist marked as menu is the menu
+    const auto& menu_it{std::ranges::find_if(m_playedPlaylists,
+                                             [](const playlistInfo& p) { return p.mightBeMenu; })};
+
+    if (menu_it != m_playedPlaylists.end())
+    {
+      const int menuPlaylist{menu_it->playlist};
+      m_playedPlaylists.erase(m_playedPlaylists.begin(), menu_it);
+
+      // Now remove any subsequent menu playlists
+      std::erase_if(m_playedPlaylists,
+                    [&menuPlaylist](const playlistInfo& p) { return p.playlist == menuPlaylist; });
+    }
+
+    // If never reached menu or ended in menu and menu only played once then never got to feature
+    if (m_playedPlaylists.empty())
+    {
+      fileItem.GetVideoInfoTag()->m_streamDetails.Reset();
+      fileItem.SetDynPath("");
+      m_State.time = 0.0;
+    }
+    else
+    {
+      // Find the playlist that was played the longest (of those that remain)
+      const auto& it{std::ranges::max_element(m_playedPlaylists,
+                                              [](const playlistInfo& i, const playlistInfo& j)
+                                              { return i.watchedTime < j.watchedTime; })};
+
+      fileItem.GetVideoInfoTag()->m_streamDetails = it->details;
+      const int playlist{it->playlist};
+      fileItem.GetVideoInfoTag()->m_iTrack = playlist;
+      CURL url{m_item.GetDynPath()};
+      url.SetFileName(StringUtils::Format("BDMV/PLAYLIST/{:05}.mpls", playlist));
+      fileItem.SetDynPath(url.Get());
+      m_State = it->state;
+    }
+  }
+  else
```

Comment:
For me, this code definitely belongs not in this class. Why should a video player know/care about all these bluray details??? This feels wrong by design.